### ISSUE TITLE
Ensure iteration count is persisted when model is saved

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/deeplearning4j#1711]
The total number of iterations is important for learning rate and momentum schedules.

Currently this is defined in base optimizer; optimizer is not saved when the network is (only config + updaters + params are). Thus, if a model is saved + loaded and training is continued, the number of iterations is reset to zero, messing up the schedules.

Ideally, the iteration count should be defined in something like MultiLayerConfiguration and ComputationGraphConfiguration.

**Comment by AlexDBlack** [comment]
Note: this is also an issue for Spark training; currently, the iteration count will not be persisted/passed between the workers and driver. Consequently, learning rate/momentum schedules will be reset after each averaging.

**Comment by AlexDBlack** [comment]
This was fixed a long time ago, but the issue was never closed.

https://github.com/deeplearning4j/deeplearning4j/blob/master/deeplearning4j-nn/src/main/java/org/deeplearning4j/optimize/solvers/BaseOptimizer.java#L260

https://github.com/deeplearning4j/deeplearning4j/blob/master/deeplearning4j-nn/src/main/java/org/deeplearning4j/optimize/solvers/BaseOptimizer.java#L367-L377

https://github.com/deeplearning4j/deeplearning4j/blob/master/deeplearning4j-scaleout/spark/dl4j-spark/src/main/java/org/deeplearning4j/spark/impl/paramavg/ParameterAveragingTrainingMaster.java#L933-L941
